elasticai/creator/nn/fixed_point/linear/testbench.py

```diff
@@ -0,0 +1,122 @@
+from abc import abstractmethod
```

Comment:
It seems there's a lot of code duplication between this module and the conv1d layer testbench.py. In case that is correct, can we get rid of these duplications?

Reply:
I don't know how to get rid of it because the fields itself differ for the layers.

Regarding the protocol: What we could think of is getting a standard design protocol which inherits name and port. But the rest will probably need to be implemented for each layer anyway. 

Regarding the testbench: This will also vary on the chosen testbench. I am not that happy with the current way but I also can't think of a different way.

Reply:
Alternative proposal developed during in person discussion:

- remove `SimulatedLayer` from `vhdl` package
- remove binary mac and mac layer from `nn` package
- move `SimulatedLayer` from here and from `conv1d` to `vhdl` package